index.js

```diff
@@ -23,24 +32,26 @@ function readFile (file, options, callback) {
     shouldThrow = options.throws
   }
 
-  fs.readFile(file, options, function (err, data) {
-    if (err) return callback(err)
-
-    data = stripBom(data)
-
-    var obj
-    try {
-      obj = JSON.parse(data, options ? options.reviver : null)
-    } catch (err2) {
-      if (shouldThrow) {
-        err2.message = file + ': ' + err2.message
-        return callback(err2)
-      } else {
-        return callback(null, null)
+  return new Promise(function (resolve, reject) {
```

Comment:
This returns a promise regardless of whether a callback was passed or not. I'm concerned this will cause edge-case bugs as people may be relying on this function returning `undefined` when passing a callback.